--- a/net/minecraft/server/level/ServerPlayerGameMode.java
+++ b/net/minecraft/server/level/ServerPlayerGameMode.java
@@ -35,6 +_,7 @@
 import net.minecraft.world.phys.shapes.VoxelShape;
 import org.jspecify.annotations.Nullable;
 import org.slf4j.Logger;
+import spout.clientview.packetmapping.blockstate.apply.breakspeed.BlockBreakSpeedLatencyCompensation;
 
 public class ServerPlayerGameMode {
     private static final double FLIGHT_DISABLE_RANGE = 1.0;
@@ -43,11 +_,13 @@
     protected final ServerPlayer player;
     private GameType gameModeForPlayer = GameType.DEFAULT_MODE;
     private @Nullable GameType previousGameModeForPlayer;
-    private boolean isDestroyingBlock;
-    private int destroyProgressStart;
+    // Spout start - Client view / Packet mapping / Block states / Apply / Breaking speed - Keep ratio history for compensation - Make fields public
+    public boolean isDestroyingBlock;
+    public int destroyProgressStart;
     private BlockPos destroyPos = BlockPos.ZERO;
     private Direction destroyDirection = Direction.DOWN;
-    private int gameTicks;
+    public int gameTicks;
+    // Spout end - Client view / Packet mapping / Block states / Apply / Breaking speed - Keep ratio history for compensation - Make fields public
     public boolean hasDelayedDestroy;
     private BlockPos delayedDestroyPos = BlockPos.ZERO;
     private int delayedTickStart;
@@ -305,6 +_,7 @@
                     BlockState state = this.level.getBlockState(pos);
                     if (!state.isAir()) {
                         float destroyProgress = state.getDestroyProgress(this.player, this.player.level(), pos) * (ticksSpentDestroying + 1);
+                        destroyProgress *= BlockBreakSpeedLatencyCompensation.getCompensationFactor(this.player, this.player.level(), pos, state); // Spout - Client view / Packet mapping / Block states / Apply / Breaking speed - Compensate for difference in intended ratio and recently observed ratio
                         if (destroyProgress >= 0.7F) {
                             this.isDestroyingBlock = false;
                             this.level.destroyBlockProgress(this.player.getId(), pos, -1);
