--- a/net/minecraft/network/HashedStack.java
+++ b/net/minecraft/network/HashedStack.java
@@ -8,6 +_,7 @@
 import net.minecraft.network.codec.StreamCodec;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
+import spout.clientview.model.ClientViewImpl;
 
 public interface HashedStack {
     HashedStack EMPTY = new HashedStack() {
@@ -21,7 +_,24 @@
             return stack.isEmpty();
         }
     };
-    StreamCodec<RegistryFriendlyByteBuf, HashedStack> STREAM_CODEC = ByteBufCodecs.optional(HashedStack.ActualItem.STREAM_CODEC)
+    // Spout start - Client view / Packet mapping / Item stacks / Apply / Reverse - Reverse-map hashed item stacks - Wrap codec
+    StreamCodec<RegistryFriendlyByteBuf, HashedStack> STREAM_CODEC = new net.minecraft.network.codec.StreamCodec<RegistryFriendlyByteBuf, HashedStack>() {
+
+        @Override
+        public HashedStack decode(final RegistryFriendlyByteBuf buffer) {
+            HashedStack decoded = UNMAPPED_STREAM_CODEC.decode(buffer);
+            spout.clientview.packetmapping.itemstack.apply.reverse.@org.jspecify.annotations.Nullable ItemStackMappingReverser itemStackMappingReverser = ((ClientViewImpl) buffer.getClientViewOrFallback()).getItemMappingReverser();
+            return itemStackMappingReverser != null ? itemStackMappingReverser.reverseMappingIfPossible(decoded) : decoded;
+        }
+
+        @Override
+        public void encode(final RegistryFriendlyByteBuf buffer, final HashedStack value) {
+            UNMAPPED_STREAM_CODEC.encode(buffer, value);
+        }
+
+    };
+    StreamCodec<RegistryFriendlyByteBuf, HashedStack> UNMAPPED_STREAM_CODEC = ByteBufCodecs.optional(HashedStack.ActualItem.STREAM_CODEC)
+        // Spout end - Client view / Packet mapping / Item stacks / Apply / Reverse - Reverse-map hashed item stacks - Wrap codec
         .map(
             actualItem -> DataFixUtils.orElse((Optional<? extends HashedStack>)actualItem, EMPTY),
             hashedStack -> hashedStack instanceof HashedStack.ActualItem actualItem ? Optional.of(actualItem) : Optional.empty()
