--- a/net/minecraft/world/item/Item.java
+++ b/net/minecraft/world/item/Item.java
@@ -109,11 +_,11 @@
 import org.jspecify.annotations.Nullable;
 import org.slf4j.Logger;
 
-public class Item implements ItemLike, FeatureElement {
+public class Item implements FeatureElement, ItemLike {
     public static final Codec<Holder<Item>> CODEC = BuiltInRegistries.ITEM
         .holderByNameCodec()
         .validate(item -> item.is(Items.AIR.builtInRegistryHolder()) ? DataResult.error(() -> "Item must not be minecraft:air") : DataResult.success(item));
-    public static final StreamCodec<RegistryFriendlyByteBuf, Holder<Item>> STREAM_CODEC = ByteBufCodecs.holderRegistry(Registries.ITEM);
+    public static final spout.clientview.packetmapping.itemstack.ItemHolderRegistryCodec STREAM_CODEC = (spout.clientview.packetmapping.itemstack.ItemHolderRegistryCodec) ByteBufCodecs.holderRegistry(Registries.ITEM); // Spout - Item mapping - Wrap codec for items
     public static final Codec<Holder<Item>> CODEC_WITH_BOUND_COMPONENTS = CODEC.validate(
         item -> !item.areComponentsBound()
             ? DataResult.error(() -> "Item " + item.getRegisteredName() + " does not have components yet")
@@ -136,6 +_,71 @@
         return item == null ? 0 : BuiltInRegistries.ITEM.getId(item);
     }
 
+    // Spout start - More data-driven - Minecraft registries - Cache block and item values
+    /**
+     * The {@link net.minecraft.resources.Identifier} with which this item is registered in
+     * {@link BuiltInRegistries#ITEM}.
+     *
+     * <p>
+     * This will be null until initialized in
+     * {@link spout.gamecontent.datadriven.item.registry.ItemRegistry#register}.
+     * </p>
+     */
+    public net.minecraft.resources.Identifier keyInItemRegistry = null;
+
+    /**
+     * The {@linkplain net.minecraft.core.Registry#getId index} at which this item is registered in
+     * {@link BuiltInRegistries#ITEM}.
+     *
+     * <p>
+     * This will be -1 until initialized in
+     * {@link spout.gamecontent.datadriven.item.registry.ItemRegistry#register}.
+     * </p>
+     */
+    public int indexInItemRegistry = -1;
+
+    /**
+     * The {@linkplain net.minecraft.core.Registry#getId index} at which this item is registered in
+     * {@link spout.gamecontent.datadriven.item.registry.VanillaOnlyItemRegistry}.
+     *
+     * <p>
+     * This will be -1 until initialized in
+     * {@link spout.gamecontent.datadriven.item.registry.VanillaOnlyItemRegistry#add}.
+     * </p>
+     */
+    public int indexInVanillaOnlyItemRegistry = -1;
+    // Spout end - More data-driven - Minecraft registries - Cache block and item values
+
+    // Spout start - More data-driven - Vanilla flag - Implementation part
+    private boolean isVanilla;
+
+    /**
+     * Returns whether this item is present in vanilla.
+     *
+     * <p>
+     * This does not take into account whether any details of this item, such as its max stack size,
+     * are different (were modified) from vanilla.
+     * </p>
+     *
+     * @return Whether this item is present in vanilla.
+     */
+    public boolean isVanilla() {
+        return this.isVanilla;
+    }
+
+    /**
+     * Sets {@link #isVanilla} to true.
+     *
+     * <p>
+     * This must only be called at most once,
+     * before the item is registered to {@link BuiltInRegistries#ITEM}.
+     * </p>
+     */
+    public void markAsVanilla() {
+        this.isVanilla = true;
+    }
+    // Spout end - More data-driven - Vanilla flag - Implementation part
+
     public static Item byId(final int id) {
         return BuiltInRegistries.ITEM.byId(id);
     }
@@ -389,15 +_,20 @@
         return false;
     }
 
-    public static class Properties {
+    // Spout start - More data-driven - Minecraft registries - Item - Implement interfaces
+    public static class Properties implements spout.gamecontent.datadriven.item.ItemPropertiesSerializationDecorator {
+        private @Nullable DataComponentMap initializedComponents = DataComponentMap.EMPTY;
+        @Override public @Nullable DataComponentMap spout$initializedComponents() { return this.initializedComponents; }
+        @Override public void spout$initializedComponents(@Nullable DataComponentMap initializedComponents) { this.initializedComponents = initializedComponents; }
+        // Spout end - More data-driven - Minecraft registries - Item - Implement interfaces
         private static final DependantName<Item, String> BLOCK_DESCRIPTION_ID = id -> Util.makeDescriptionId("block", id.identifier());
         private static final DependantName<Item, String> ITEM_DESCRIPTION_ID = id -> Util.makeDescriptionId("item", id.identifier());
-        private DataComponentInitializers.Initializer<Item> componentInitializer = (builder, var1, var2) -> builder.addAll(
+        public DataComponentInitializers.Initializer<Item> componentInitializer = (builder, var1, var2) -> builder.addAll( // Spout - Common class tweakers - Make accessible
             DataComponents.COMMON_ITEM_COMPONENTS
         );
         private @Nullable ItemStackTemplate craftingRemainingItem;
-        private FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET;
-        private @Nullable ResourceKey<Item> id;
+        public FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET; // Spout - Common class tweakers - Make accessible
+        public @Nullable ResourceKey<Item> id; // Spout - Common class tweakers - Make accessible
         private DependantName<Item, String> descriptionId = ITEM_DESCRIPTION_ID;
         private final DependantName<Item, Identifier> model = ResourceKey::identifier;
 
