--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
@@ -136,10 +_,12 @@
     }
 
     // ========================================================================
-    private static final Map<Block, Material> BLOCK_MATERIAL = new IdentityHashMap<>();
-    private static final Map<Item, Material> ITEM_MATERIAL = new IdentityHashMap<>();
-    private static final Map<Material, Item> MATERIAL_ITEM = new EnumMap<>(Material.class);
-    private static final Map<Material, Block> MATERIAL_BLOCK = new EnumMap<>(Material.class);
+    // Spout start - Material injection - Implementation part - Make public and don't use enum maps
+    public static final Map<Block, Material> BLOCK_MATERIAL = new IdentityHashMap<>();
+    public static final Map<Item, Material> ITEM_MATERIAL = new IdentityHashMap<>();
+    public static final Map<Material, Item> MATERIAL_ITEM = new IdentityHashMap<>();
+    public static final Map<Material, Block> MATERIAL_BLOCK = new IdentityHashMap<>();
+    // Spout end - Material injection - Implementation part - Make public and don't use enum maps
 
     static {
         for (Block block : BuiltInRegistries.BLOCK) {
