--- a/net/minecraft/world/level/block/RespawnAnchorBlock.java
+++ b/net/minecraft/world/level/block/RespawnAnchorBlock.java
@@ -219,7 +_,7 @@
         return true;
     }
 
-    public static int getScaledChargeLevel(final BlockState state, final int maximum) {
+    public static int getScaledChargeLevel(spout.common.moredatadriven.minecraft.type.BlockStateFunction.ByProperties.PropertyValueMap state, int maximum) { // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state function properties serializable
         return Mth.floor((state.getValue(CHARGE) - 0) / 4.0F * maximum);
     }
 
