--- a/net/minecraft/network/protocol/game/ClientboundSectionBlocksUpdatePacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundSectionBlocksUpdatePacket.java
@@ -68,7 +_,13 @@
         output.writeVarInt(this.positions.length);
 
         for (int i = 0; i < this.positions.length; i++) {
-            output.writeVarLong((long)Block.getId(this.states[i]) << 12 | this.positions[i]);
+            // Spout start - Block mapping - Implementation part - Apply during section blocks update write
+            int x = this.sectionPos.relativeToBlockX(this.positions[i]);
+            int y = this.sectionPos.relativeToBlockY(this.positions[i]);
+            int z = this.sectionPos.relativeToBlockZ(this.positions[i]);
+            spout.server.paper.api.packetmapping.block.BlockMappingFunctionContext context = new spout.server.paper.impl.packetmapping.block.BlockMappingFunctionContextImpl(output.getClientViewOrFallback(), true, x, y, z);
+            output.writeVarLong((long)Block.getId(spout.server.paper.impl.packetmapping.block.BlockMappingsImpl.get().apply(this.states[i], context)) << 12 | this.positions[i]);
+            // Spout end - Block mapping - Implementation part - Apply during section blocks update write
         }
     }
 
