--- a/src/main/java/io/papermc/paper/registry/entry/RegistryEntryMeta.java
+++ b/src/main/java/io/papermc/paper/registry/entry/RegistryEntryMeta.java
@@ -113,6 +_,11 @@
         @Override
         public org.bukkit.Registry<A> createApiRegistry(final Registry<M> nmsRegistry) {
             if (this.modificationApiSupport.canAdd()) {
+                // Spout start - More data-driven - Paper registry API - Implementation part - Make block and item registries key-aware
+                if (nmsRegistry instanceof spout.server.paper.impl.moredatadriven.minecraft.BlockRegistry || nmsRegistry instanceof spout.server.paper.impl.moredatadriven.minecraft.ItemRegistry) {
+                    return (org.bukkit.Registry<A>) new spout.server.paper.impl.moredatadriven.paper.registry.KeyAwareWritableCraftRegistry((MappedRegistry<M>) nmsRegistry, this);
+                }
+                // Spout end - More data-driven - Paper registry API - Implementation part - Make block and item registries key-aware
                 return new WritableCraftRegistry<>((MappedRegistry<M>) nmsRegistry, this);
             } else {
                 return ServerSide.super.createApiRegistry(nmsRegistry);
