--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBlockType.java
+++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBlockType.java
@@ -251,4 +_,11 @@
         return this.getHandle().hasCollision;
     }
     // Paper end - hasCollision API
+
+    // Spout start - More data-driven - Vanilla flag - Implementation part
+    @Override
+    public boolean isVanilla() {
+        return this.getHandle().isVanilla();
+    }
+    // Spout end - More data-driven - Vanilla flag - Implementation part
 }
