--- a/net/minecraft/world/level/block/Blocks.java
+++ b/net/minecraft/world/level/block/Blocks.java
@@ -49,12 +_,12 @@
 import net.minecraft.world.phys.AABB;
 
 public class Blocks {
-    private static final BlockBehaviour.StatePredicate NOT_CLOSED_SHULKER = (state, level, pos) -> !(
+    public static final BlockBehaviour.StatePredicate NOT_CLOSED_SHULKER = (state, level, pos) -> !( // Spout - Common class tweakers - Make accessible
             level.getBlockEntity(pos) instanceof ShulkerBoxBlockEntity shulkerBoxBlockEntity
         )
         || shulkerBoxBlockEntity.isClosed();
-    private static final BlockBehaviour.StatePredicate NOT_EXTENDED_PISTON = (state, level, pos) -> !state.getValue(PistonBaseBlock.EXTENDED);
-    private static final BlockBehaviour.StateArgumentPredicate<AABB> NEAR_PLANE_INTERSECTS_OUTLINE = (state, level, blockPos, nearPlaneBox) -> {
+    public static final BlockBehaviour.StatePredicate NOT_EXTENDED_PISTON = (state, level, pos) -> !state.getValue(PistonBaseBlock.EXTENDED); // Spout - Common class tweakers - Make accessible
+    public static final BlockBehaviour.StateArgumentPredicate<AABB> NEAR_PLANE_INTERSECTS_OUTLINE = (state, level, blockPos, nearPlaneBox) -> { // Spout - Common class tweakers - Make accessible
         for (AABB outlineBox : state.getOcclusionShape().toAabbs()) {
             if (outlineBox.move(blockPos).intersects(nearPlaneBox)) {
                 return true;
@@ -2138,7 +_,7 @@
             .strength(0.1F)
             .requiresCorrectToolForDrops()
             .sound(SoundType.SNOW)
-            .isViewBlocking((state, level, pos, aabb) -> state.getValue(SnowLayerBlock.LAYERS) >= 8)
+            .isViewBlocking(new spout.gamecontent.datadriven.block.subtypes.KnownStateAABBPredicate.Via(spout.gamecontent.datadriven.block.subtypes.KnownStatePredicate.MAX_SNOW_LAYERS)) // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
             .pushReaction(PushReaction.POPPED)
     );
     public static final Block ICE = register(
@@ -5780,7 +_,7 @@
         return blockState -> blockState.getValue(BlockStateProperties.WATERLOGGED) ? MapColor.WATER : mapColor;
     }
 
-    private static Boolean never(final BlockState state, final BlockGetter blockGetter, final BlockPos blockPos, final EntityType<?> entityType) {
+    public static Boolean never(BlockState state, BlockGetter level, BlockPos pos, EntityType<?> entity) { // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Defaults for types - Make public
         return false;
     }
 
@@ -5788,7 +_,7 @@
         return true;
     }
 
-    private static Boolean ocelotOrParrot(final BlockState state, final BlockGetter blockGetter, final BlockPos blockPos, final EntityType<?> entityType) {
+    public static Boolean ocelotOrParrot(final BlockState state, final BlockGetter blockGetter, final BlockPos blockPos, final EntityType<?> entityType) { // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Defaults for types - Make public
         return entityType == EntityTypes.OCELOT || entityType == EntityTypes.PARROT;
     }
 
@@ -5802,14 +_,14 @@
     }
 
     private static BlockBehaviour.Properties netherStemProperties(final MapColor mapColor) {
-        return BlockBehaviour.Properties.of().mapColor(state -> mapColor).instrument(NoteBlockInstrument.BASS).strength(2.0F).sound(SoundType.STEM);
+        return BlockBehaviour.Properties.of().mapColor(mapColor).instrument(NoteBlockInstrument.BASS).strength(2.0F).sound(SoundType.STEM); // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make map color property serializable
     }
 
-    private static boolean always(final BlockState state, final BlockGetter blockGetter, final BlockPos blockPos) {
+    public static boolean always(BlockState state, BlockGetter level, BlockPos pos) { // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make public
         return true;
     }
 
-    private static boolean never(final BlockState state, final BlockGetter blockGetter, final BlockPos blockPos) {
+    public static boolean never(BlockState state, BlockGetter level, BlockPos pos) { // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make public
         return false;
     }
 
@@ -5924,6 +_,13 @@
         final ResourceKey<Block> id, final Function<BlockBehaviour.Properties, Block> factory, final BlockBehaviour.Properties properties
     ) {
         Block block = factory.apply(properties.setId(id));
+        // Spout start - More data-driven - Vanilla flag - Implementation part
+        block.markAsVanilla();
+        // Currently, there is no way to modify block states of vanilla blocks, so we assume all block states of vanilla blocks are vanilla too
+        for (BlockState blockState : block.getStateDefinition().getPossibleStates()) {
+            blockState.markAsVanilla();
+        }
+        // Spout end - More data-driven - Vanilla flag - Implementation part
         return Registry.register(BuiltInRegistries.BLOCK, id, block);
     }
 
@@ -5943,7 +_,7 @@
             for (Block block : BuiltInRegistries.BLOCK) {
                 boolean first = true;
                 for (BlockState state : block.getStateDefinition().getPossibleStates()) {
-                    Block.BLOCK_STATE_REGISTRY.add(state);
+                    // Block.BLOCK_STATE_REGISTRY.add(state); // Spout - More data-driven - Minecraft registries - Register block states in BlockRegistry#register
                     if (first) {
                         first = false;
                         state.initCache();
@@ -5955,6 +_,7 @@
             // Make sure this stays as a method reference, else the synthetic lambda method makes this blow up
             statesToInit.parallelStream().forEach(BlockBehaviour.BlockStateBase::initCache);
             // Paper end - Perf: Optimize block state cache initialization
+            spout.gamecontent.datadriven.block.registry.BlockRegistry.get().canInitializeBlockStateCaches = true; // Spout - More data-driven - Minecraft registries - Initialize block state caches for non-vanilla blocks in BlockRegistry#register
         });
     }
 }
