--- a/net/minecraft/world/level/block/FarmlandBlock.java
+++ b/net/minecraft/world/level/block/FarmlandBlock.java
@@ -27,12 +_,12 @@
 import org.jspecify.annotations.Nullable;
 
 public class FarmlandBlock extends Block {
-    private final Block baseBlock;
+    public final Block baseBlock; // Spout - Common class tweakers - Make accessible
     public static final IntegerProperty MOISTURE = BlockStateProperties.MOISTURE;
     private static final VoxelShape SHAPE = Block.column(16.0, 0.0, 15.0);
     public static final int MAX_MOISTURE = 7;
 
-    protected FarmlandBlock(final Block baseBlock, final BlockBehaviour.Properties properties) {
+    public FarmlandBlock(final Block baseBlock, final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
         this.registerDefaultState(this.stateDefinition.any().setValue(MOISTURE, 0));
         this.baseBlock = baseBlock;
