--- a/net/minecraft/world/level/block/FenceGateBlock.java
+++ b/net/minecraft/world/level/block/FenceGateBlock.java
@@ -51,7 +_,7 @@
     private static final Map<Direction.Axis, VoxelShape> SHAPE_OCCLUSION_WALL = Maps.newEnumMap(
         Util.mapValues(SHAPE_OCCLUSION, v -> v.move(0.0, -0.1875, 0.0).optimize())
     );
-    private final WoodType type;
+    public final WoodType type; // Spout - Common class tweakers - Make accessible
 
     public FenceGateBlock(final WoodType type, final BlockBehaviour.Properties properties) {
         super(properties.sound(type.soundType()));
