--- a/net/minecraft/world/level/block/FlowerPotBlock.java
+++ b/net/minecraft/world/level/block/FlowerPotBlock.java
@@ -30,7 +_,7 @@
 public class FlowerPotBlock extends Block {
     private static final Map<Block, Block> POTTED_BY_CONTENT = Maps.newHashMap();
     private static final VoxelShape SHAPE = Block.column(6.0, 0.0, 6.0);
-    private final Block potted;
+    public final Block potted; // Spout - Common class tweakers - Make accessible
 
     public FlowerPotBlock(final Block potted, final BlockBehaviour.Properties properties) {
         super(properties);
