--- a/net/minecraft/world/level/block/RedstoneWallTorchBlock.java
+++ b/net/minecraft/world/level/block/RedstoneWallTorchBlock.java
@@ -25,7 +_,7 @@
     public static final EnumProperty<Direction> FACING = HorizontalDirectionalBlock.FACING;
     public static final BooleanProperty LIT = RedstoneTorchBlock.LIT;
 
-    protected RedstoneWallTorchBlock(final BlockBehaviour.Properties properties) {
+    public RedstoneWallTorchBlock(final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
         this.registerDefaultState(this.stateDefinition.any().setValue(FACING, Direction.NORTH).setValue(LIT, true));
     }
