--- a/net/minecraft/world/level/block/SmithingTableBlock.java
+++ b/net/minecraft/world/level/block/SmithingTableBlock.java
@@ -17,7 +_,7 @@
 public class SmithingTableBlock extends CraftingTableBlock {
     private static final Component CONTAINER_TITLE = Component.translatable("container.upgrade");
 
-    protected SmithingTableBlock(final BlockBehaviour.Properties properties) {
+    public SmithingTableBlock(final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
     }
 
