--- a/net/minecraft/world/level/block/StairBlock.java
+++ b/net/minecraft/world/level/block/StairBlock.java
@@ -39,9 +_,9 @@
     private static final Map<Direction, VoxelShape> SHAPE_TOP_STRAIGHT = Shapes.rotateHorizontal(SHAPE_STRAIGHT, OctahedralGroup.INVERT_Y);
     private static final Map<Direction, VoxelShape> SHAPE_TOP_INNER = Shapes.rotateHorizontal(SHAPE_INNER, OctahedralGroup.INVERT_Y);
     private final Block base;
-    protected final BlockState baseState;
+    public final BlockState baseState; // Spout - Common class tweakers - Make accessible
 
-    protected StairBlock(final BlockState baseState, final BlockBehaviour.Properties properties) {
+    public StairBlock(final BlockState baseState, final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
         this.registerDefaultState(
             this.stateDefinition
