--- a/net/minecraft/world/level/block/WaterloggedTransparentBlock.java
+++ b/net/minecraft/world/level/block/WaterloggedTransparentBlock.java
@@ -18,7 +_,7 @@
 public class WaterloggedTransparentBlock extends TransparentBlock implements SimpleWaterloggedBlock {
     public static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
 
-    protected WaterloggedTransparentBlock(final BlockBehaviour.Properties properties) {
+    public WaterloggedTransparentBlock(final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
         this.registerDefaultState(this.defaultBlockState().setValue(WATERLOGGED, false));
     }
