--- a/net/minecraft/world/level/block/WeatheringCopperGrateBlock.java
+++ b/net/minecraft/world/level/block/WeatheringCopperGrateBlock.java
@@ -9,7 +_,7 @@
 public class WeatheringCopperGrateBlock extends WaterloggedTransparentBlock implements WeatheringCopper {
     private final WeatheringCopper.WeatherState weatherState;
 
-    protected WeatheringCopperGrateBlock(final WeatheringCopper.WeatherState weatherState, final BlockBehaviour.Properties properties) {
+    public WeatheringCopperGrateBlock(final WeatheringCopper.WeatherState weatherState, final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
         this.weatherState = weatherState;
     }
