--- a/net/minecraft/world/level/block/WeatheringLanternBlock.java
+++ b/net/minecraft/world/level/block/WeatheringLanternBlock.java
@@ -9,7 +_,7 @@
 public class WeatheringLanternBlock extends LanternBlock implements WeatheringCopper {
     private final WeatheringCopper.WeatherState weatherState;
 
-    protected WeatheringLanternBlock(final WeatheringCopper.WeatherState weatherState, final BlockBehaviour.Properties properties) {
+    public WeatheringLanternBlock(final WeatheringCopper.WeatherState weatherState, final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
         this.weatherState = weatherState;
     }
