--- a/net/minecraft/world/level/block/WetSpongeBlock.java
+++ b/net/minecraft/world/level/block/WetSpongeBlock.java
@@ -12,7 +_,7 @@
 import net.minecraft.world.level.block.state.BlockState;
 
 public class WetSpongeBlock extends Block {
-    protected WetSpongeBlock(final BlockBehaviour.Properties properties) {
+    public WetSpongeBlock(final BlockBehaviour.Properties properties) { // Spout - Common class tweakers - Make accessible
         super(properties);
     }
 
