--- a/net/minecraft/network/Connection.java
+++ b/net/minecraft/network/Connection.java
@@ -54,7 +_,16 @@
 import org.slf4j.Marker;
 import org.slf4j.MarkerFactory;
 
-public class Connection extends SimpleChannelInboundHandler<Packet<?>> {
+public class Connection extends SimpleChannelInboundHandler<Packet<?>> implements spout.clientview.packetmapping.clientviewlookup.ClientViewLookup { // Spout - Client view - Lookup - Implement for Connection
+    public spout.clientview.model.@Nullable ClientView clientView; // Spout - Client view - Set - Field
+    // Spout start - Client view - Lookup - Implement for Connection - Getter
+    @Override
+    public spout.clientview.model.@Nullable ClientView getClientView() {
+        return this.clientView;
+    }
+    // Spout end - Client view - Lookup - Implement for Connection - Getter
+    public final spout.server.paper.impl.clientview.set.ClientViewSetter clientViewSetter; // Spout - Client view - Set - Setter
+    public final spout.clientview.packetmapping.itemstack.apply.reverse.ItemStackMappingReverser itemStackMappingReverser; // Spout - Client view / Packet mapping / Item stacks / Apply / Reverse - Store reverser in Connection
     private static final float AVERAGE_PACKETS_SMOOTHING = 0.75F;
     private static final Logger LOGGER = LogUtils.getLogger();
     public static final Marker ROOT_MARKER = MarkerFactory.getMarker("NETWORK");
@@ -129,12 +_,15 @@
 
     public Connection(final PacketFlow receiving) {
         this.receiving = receiving;
+        this.clientViewSetter = new spout.server.paper.impl.clientview.set.ClientViewSetter(this); // Spout - Client view - Set - Setter
+        this.itemStackMappingReverser = new spout.clientview.packetmapping.itemstack.apply.reverse.ItemStackMappingReverser(); // Spout - Client view / Packet mapping / Item stacks / Apply / Reverse - Store reverser in Connection
     }
 
     @Override
     public void channelActive(final ChannelHandlerContext ctx) throws Exception {
         super.channelActive(ctx);
         this.channel = ctx.channel();
+        spout.clientview.packetmapping.clientviewlookup.NettyChannelClientViewLookup.set(this.channel, this); // Spout - Client view - Packet mapping - Client view lookup - Netty channel - Copy the connection (acting as the lookup) to the channel
         this.address = this.channel.remoteAddress();
         this.preparing = false; // Spigot
         if (this.delayedDisconnect != null) {
