--- a/net/minecraft/world/level/Level.java
+++ b/net/minecraft/world/level/Level.java
@@ -170,6 +_,12 @@
         return this.paperConfig;
     }
     // Paper end - add paper world config
+    // Spout start - Configuration - World configuration
+    private final spout.server.paper.impl.configuration.SpoutWorldConfiguration spoutConfig;
+    public spout.server.paper.impl.configuration.SpoutWorldConfiguration spoutConfig() {
+        return this.spoutConfig;
+    }
+    // Spout end - Configuration - World configuration
 
     public static @Nullable BlockPos lastPhysicsProblem; // Spigot
     public final Map<ServerExplosion.CacheKey, Float> explosionDensityCache = new java.util.HashMap<>(); // Paper - Optimize explosions
@@ -838,6 +_,7 @@
         org.bukkit.World.Environment environment, // Paper
         java.util.function.Function<org.spigotmc.SpigotWorldConfig, // Spigot - create per world config
         io.papermc.paper.configuration.WorldConfiguration> paperWorldConfigCreator, // Paper - create paper world config
+        java.util.function.Function<org.spigotmc.SpigotWorldConfig, spout.server.paper.impl.configuration.SpoutWorldConfiguration> spoutWorldConfigCreator, // Spout - Configuration - Create world configuration
         java.util.concurrent.Executor executor // Paper - Anti-Xray
     ) {
         // Paper start - getblock optimisations - cache world height/sections
@@ -852,6 +_,7 @@
         final org.bukkit.NamespacedKey worldKey = CraftNamespacedKey.fromMinecraft(dimension.identifier()); // Paper
         this.spigotConfig = new org.spigotmc.SpigotWorldConfig(bukkitName, worldKey); // Spigot
         this.paperConfig = paperWorldConfigCreator.apply(this.spigotConfig); // Paper - create paper world config
+        this.spoutConfig = spoutWorldConfigCreator.apply(this.spigotConfig); // Spout - Configuration - Create world configuration
         this.generator = generator;
         this.world = new CraftWorld((ServerLevel) this, worldKey, biomeProvider, environment);
 
