--- a/net/minecraft/server/Main.java
+++ b/net/minecraft/server/Main.java
@@ -104,6 +_,18 @@
                 JvmProfiler.INSTANCE.start(Environment.SERVER);
             }
 
+            // Spout start - EULA before bootstrap - Add before bootstrap
+            Path eulaFile = Paths.get("eula.txt");
+            Eula eula = new Eula(eulaFile);
+            // Spout end - EULA before bootstrap - Add before bootstrap
+
+            // Spout start - Configuration - Load global configuration before bootstrap
+            var spoutConfigurations = spout.server.paper.impl.configuration.SpoutConfigurations.get(options);
+            spoutConfigurations.initializeGlobalConfiguration(null);
+            // Spout end - Configuration - Load global configuration before bootstrap
+
+            if (!spout.server.paper.impl.confirmpermanence.ConfirmPermanence.check(LOGGER)) return; // Spout - Confirm permanence - Check and exit if necessary
+
             io.papermc.paper.plugin.PluginInitializerManager.load(options); // Paper
             Bootstrap.bootStrap();
             Bootstrap.validate();
@@ -118,8 +_,12 @@
             }
             // Paper end
             RegionFileVersion.configure(settings.getProperties().regionFileComression);
-            Path eulaFile = Paths.get("eula.txt");
-            Eula eula = new Eula(eulaFile);
+
+            // Spout start - EULA before bootstrap - Remove original
+            // Path eulaFile = Paths.get("eula.txt");
+            // Eula eula = new Eula(eulaFile);
+            // Spout end - EULA before bootstrap - Remove original
+
             // Paper start - load config files early for access below if needed
             org.bukkit.configuration.file.YamlConfiguration bukkitConfiguration = io.papermc.paper.configuration.PaperConfigurations.loadLegacyConfigFile((File) options.valueOf("bukkit-settings"));
             org.bukkit.configuration.file.YamlConfiguration spigotConfiguration = io.papermc.paper.configuration.PaperConfigurations.loadLegacyConfigFile((File) options.valueOf("spigot-settings"));
