--- a/net/minecraft/world/level/block/state/BlockBehaviour.java
+++ b/net/minecraft/world/level/block/state/BlockBehaviour.java
@@ -82,7 +_,10 @@
 import net.minecraft.world.phys.shapes.VoxelShape;
 import org.jspecify.annotations.Nullable;
 
-public abstract class BlockBehaviour implements FeatureElement {
+// Spout start - More data-driven - Minecraft registries - Subtypes - Implement interfaces
+public abstract class BlockBehaviour implements FeatureElement, spout.gamecontent.datadriven.block.subtypes.BlockBehaviourMaxVerticalOffsetAccessor {
+    @Override public float spout$getMaxVerticalOffset() { return this.getMaxVerticalOffset(); }
+    // Spout end - More data-driven - Minecraft registries - Subtypes - Implement interfaces
     protected static final Direction[] UPDATE_SHAPE_ORDER = new Direction[]{
         Direction.WEST, Direction.EAST, Direction.NORTH, Direction.SOUTH, Direction.DOWN, Direction.UP
     };
@@ -352,13 +_,21 @@
     }
 
     protected float getDestroyProgress(final BlockState state, final Player player, final BlockGetter level, final BlockPos pos) {
+        // Spout start - Client view / Packet mapping / Block states / Apply / Breaking speed - Implement calculating destroy progress with given held item stack
+        return this.getDestroyProgress(state, player, level, pos, null);
+    }
+
+    protected float getDestroyProgress(final BlockState state, final Player player, final BlockGetter level, final BlockPos pos, @Nullable ItemStack heldItemStack) {
+        // Spout end - Client view / Packet mapping / Block states / Apply / Breaking speed - Implement calculating destroy progress with given held item stack
         float destroySpeed = state.getDestroySpeed(level, pos);
         if (destroySpeed == -1.0F) {
             return 0.0F;
         }
 
-        int modifier = player.hasCorrectToolForDrops(state) ? 30 : 100;
-        return player.getDestroySpeed(state) / destroySpeed / modifier;
+        // Spout start - Client view / Packet mapping / Block states / Apply / Breaking speed - Implement calculating destroy progress with given held item stack
+        int modifier = player.hasCorrectToolForDrops(state, heldItemStack) ? 30 : 100;
+        return player.getDestroySpeed(state, heldItemStack) / destroySpeed / modifier;
+        // Spout end - Client view / Packet mapping / Block states / Apply / Breaking speed - Implement calculating destroy progress with given held item stack
     }
 
     protected void spawnAfterBreak(final BlockState state, final ServerLevel level, final BlockPos pos, final ItemStack tool, final boolean dropExperience) {
@@ -450,7 +_,7 @@
         private boolean legacySolid;
         private final PushReaction pushReaction;
         private final MapColor mapColor;
-        private final float destroySpeed;
+        public final float destroySpeed; // Spout - Client view / Packet mapping / Block states / Macros / Processor - Valid proxy candidate function - Make public
         private final boolean requiresCorrectToolForDrops;
         private final boolean canOcclude;
         private final BlockBehaviour.StatePredicate isRedstoneConductor;
@@ -783,7 +_,13 @@
         }
 
         public float getDestroyProgress(final Player player, final BlockGetter level, final BlockPos pos) {
-            return this.getBlock().getDestroyProgress(this.asState(), player, level, pos);
+            // Spout start - Client view / Packet mapping / Block states / Apply / Breaking speed - Implement calculating destroy progress with given held item stack
+            return this.getDestroyProgress(player, level, pos, null);
+        }
+
+        public float getDestroyProgress(Player player, BlockGetter level, BlockPos pos, @Nullable ItemStack heldItemStack) {
+            return this.getBlock().getDestroyProgress(this.asState(), player, level, pos, heldItemStack);
+            // Spout end - Client view / Packet mapping / Block states / Apply / Breaking speed - Implement calculating destroy progress with given held item stack
         }
 
         public int getDirectSignal(final BlockGetter level, final BlockPos pos, final Direction direction) {
@@ -1140,49 +_,49 @@
     }
 
     public static class Properties {
-        private Function<BlockState, MapColor> mapColor = state -> MapColor.NONE;
-        private boolean hasCollision = true;
-        private SoundType soundType = SoundType.STONE;
-        private ToIntFunction<BlockState> lightEmission = state -> 0;
-        private float explosionResistance;
-        private float destroyTime;
-        private boolean requiresCorrectToolForDrops;
-        private boolean isRandomlyTicking;
-        private float friction = 0.6F;
-        private float speedFactor = 1.0F;
-        private float jumpFactor = 1.0F;
-        private float bounceRestitution;
-        private @Nullable ResourceKey<Block> id;
+        public static final com.mojang.serialization.Codec<net.minecraft.world.level.block.state.BlockBehaviour.Properties> CODEC = spout.gamecontent.datadriven.block.BlockPropertiesCodec.CODEC; // Spout - More data-driven - Minecraft registries - Block - Add missing properties codec implementation
+        public Function<BlockState, MapColor> mapColor = state -> MapColor.NONE; // Spout - Common class tweakers - Make accessible
+        public boolean hasCollision = true; // Spout - Common class tweakers - Make accessible
+        public SoundType soundType = SoundType.STONE; // Spout - Common class tweakers - Make accessible
+        public ToIntFunction<BlockState> lightEmission = state -> 0; // Spout - Common class tweakers - Make accessible
+        public float explosionResistance; // Spout - Common class tweakers - Make accessible
+        public float destroyTime; // Spout - Common class tweakers - Make accessible
+        public boolean requiresCorrectToolForDrops; // Spout - Common class tweakers - Make accessible
+        public boolean isRandomlyTicking; // Spout - Common class tweakers - Make accessible
+        public float friction = 0.6F; // Spout - Common class tweakers - Make accessible
+        public float speedFactor = 1.0F; // Spout - Common class tweakers - Make accessible
+        public float jumpFactor = 1.0F; // Spout - Common class tweakers - Make accessible
+        public float bounceRestitution; // Spout - Common class tweakers - Make accessible
+        public @Nullable ResourceKey<Block> id; // Spout - Common class tweakers - Make accessible
         private DependantName<Block, Optional<ResourceKey<LootTable>>> drops = id -> Optional.of(
             ResourceKey.create(Registries.LOOT_TABLE, id.identifier().withPrefix("blocks/"))
         );
         private DependantName<Block, String> descriptionId = id -> Util.makeDescriptionId("block", id.identifier());
-        private boolean canOcclude = true;
-        private boolean isAir;
-        private boolean ignitedByLava;
-        @Deprecated
-        private boolean liquid;
-        @Deprecated
-        private boolean forceSolidOff;
-        private boolean forceSolidOn;
-        private PushReaction pushReaction = PushReaction.PUSH_PULL;
-        private boolean spawnTerrainParticles = true;
-        private NoteBlockInstrument instrument = NoteBlockInstrument.HARP;
-        private boolean replaceable;
+        public boolean canOcclude = true; // Spout - Common class tweakers - Make accessible
+        public boolean isAir; // Spout - Common class tweakers - Make accessible
+        public boolean ignitedByLava; // Spout - Common class tweakers - Make accessible
+        @Deprecated
+        public boolean liquid; // Spout - Common class tweakers - Make accessible
+        @Deprecated
+        public boolean forceSolidOff; // Spout - Common class tweakers - Make accessible
+        public boolean forceSolidOn; // Spout - Common class tweakers - Make accessible
+        public PushReaction pushReaction = PushReaction.PUSH_PULL; // Spout - Common class tweakers - Make accessible
+        public boolean spawnTerrainParticles = true; // Spout - Common class tweakers - Make accessible
+        public NoteBlockInstrument instrument = NoteBlockInstrument.HARP; // Spout - Common class tweakers - Make accessible
+        public boolean replaceable; // Spout - Common class tweakers - Make accessible
         private BlockBehaviour.StateArgumentPredicate<EntityType<?>> isValidSpawn = (state, level, pos, entityType) -> state.isFaceSturdy(
                 level, pos, Direction.UP
             )
             && state.getLightEmission() < 14;
-        private BlockBehaviour.StatePredicate isRedstoneConductor = BlockBehaviour.BlockStateBase::isCollisionShapeFullBlock;
-        private BlockBehaviour.StatePredicate isSuffocating = (state, level, pos) -> state.is(BlockTags.CAUSES_SUFFOCATION)
-            && state.isCollisionShapeFullBlock(level, pos);
-        private BlockBehaviour.StateArgumentPredicate<AABB> isViewBlocking = (state, level, pos, var4) -> this.isSuffocating.test(state, level, pos);
-        private BlockBehaviour.PostProcess postProcess = (state, level, pos) -> null;
-        private Predicate<BlockState> emissiveRendering = var0 -> false;
-        private boolean dynamicShape;
-        private FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET;
-        private BlockBehaviour.@Nullable OffsetFunction offsetFunction;
-        private float fallDistanceReduction;
+        public BlockBehaviour.StatePredicate isRedstoneConductor = BlockBehaviour.BlockStateBase::isCollisionShapeFullBlock; // Spout - Common class tweakers - Make accessible
+        public BlockBehaviour.StatePredicate isSuffocating = spout.gamecontent.datadriven.block.subtypes.KnownStatePredicate.CAUSES_SUFFOCATION; // Spout - Common class tweakers - Make accessible // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+        public BlockBehaviour.StateArgumentPredicate<AABB> isViewBlocking = new spout.gamecontent.datadriven.block.subtypes.KnownStateAABBPredicate.Via(spout.gamecontent.datadriven.block.subtypes.KnownStatePredicate.wrap(this.isSuffocating)); // Spout - Common class tweakers - Make accessible
+        public BlockBehaviour.PostProcess postProcess = (state, level, pos) -> null; // Spout - Common class tweakers - Make accessible
+        public Predicate<BlockState> emissiveRendering = var0 -> false; // Spout - Common class tweakers - Make accessible // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable
+        public boolean dynamicShape; // Spout - Common class tweakers - Make accessible
+        public FeatureFlagSet requiredFeatures = FeatureFlags.VANILLA_SET; // Spout - Common class tweakers - Make accessible
+        public BlockBehaviour.@Nullable OffsetFunction offsetFunction; // Spout - Common class tweakers - Make accessible
+        public float fallDistanceReduction; // Spout - Common class tweakers - Make accessible
 
         private Properties() {
         }
@@ -1476,7 +_,7 @@
     }
 
     @FunctionalInterface
-    public interface StatePredicate {
+    public interface StatePredicate extends java.io.Serializable { // Spout - More data-driven - Minecraft registries - Types - Properties serialization - Make block state predicate properties serializable - Make predicate extend serializable so that we can inspect it in KnownStatePrediate#wrap
         boolean test(BlockState state, BlockGetter level, BlockPos pos);
     }
 }
