--- a/net/minecraft/server/players/PlayerList.java
+++ b/net/minecraft/server/players/PlayerList.java
@@ -49,12 +_,10 @@
 import net.minecraft.network.protocol.game.ClientboundSetBorderSizePacket;
 import net.minecraft.network.protocol.game.ClientboundSetBorderWarningDelayPacket;
 import net.minecraft.network.protocol.game.ClientboundSetBorderWarningDistancePacket;
-import net.minecraft.network.protocol.game.ClientboundSetChunkCacheRadiusPacket;
 import net.minecraft.network.protocol.game.ClientboundSetDefaultSpawnPositionPacket;
 import net.minecraft.network.protocol.game.ClientboundSetExperiencePacket;
 import net.minecraft.network.protocol.game.ClientboundSetHeldSlotPacket;
 import net.minecraft.network.protocol.game.ClientboundSetPlayerTeamPacket;
-import net.minecraft.network.protocol.game.ClientboundSetSimulationDistancePacket;
 import net.minecraft.network.protocol.game.ClientboundSoundPacket;
 import net.minecraft.network.protocol.game.ClientboundUpdateMobEffectPacket;
 import net.minecraft.network.protocol.game.ClientboundUpdateRecipesPacket;
@@ -101,6 +_,7 @@
 import net.minecraft.world.scores.Team;
 import org.jspecify.annotations.Nullable;
 import org.slf4j.Logger;
+import spout.clientview.packetmapping.blockstate.apply.breakspeed.BlockBreakSpeedFactorUpdater;
 
 public abstract class PlayerList {
     public static final File USERBANLIST_FILE = new File("banned-players.json");
@@ -235,7 +_,7 @@
         // CraftBukkit start
         org.bukkit.craftbukkit.entity.CraftPlayer bukkitPlayer = player.getBukkitEntity();
 
-        // Ensure that player inventory is populated with its viewer
+        // Ensure that player inventory is populated with its vjiewer
         player.containerMenu.transferTo(player.containerMenu, bukkitPlayer);
 
         org.bukkit.event.player.PlayerJoinEvent playerJoinEvent = new org.bukkit.event.player.PlayerJoinEvent(bukkitPlayer, io.papermc.paper.adventure.PaperAdventure.asAdventure(component)); // Paper - Adventure
@@ -330,7 +_,7 @@
                 .getOrThrow(net.minecraft.world.level.biome.Biomes.PLAINS);
             player.connection.send(new net.minecraft.network.protocol.game.ClientboundLevelChunkWithLightPacket(
                 new net.minecraft.world.level.chunk.EmptyLevelChunk(level, player.chunkPosition(), plains),
-                level.getLightEngine(), (java.util.BitSet)null, (java.util.BitSet) null, true) // Paper - Anti-Xray
+                level.getLightEngine(), (java.util.BitSet)null, (java.util.BitSet) null, true, player) // Paper - Anti-Xray // Spout - Client view / Packet mapping / Block states / Apply / Chunk packets - Pass the player
             );
         }
         // Paper end - Send empty chunk
@@ -742,6 +_,7 @@
             // CraftBukkit end
             this.sendAllPlayerInfoIn = 0;
         }
+        BlockBreakSpeedFactorUpdater.updateFactor(this.players); // Spout - Client view / Packet mapping / Block states / Apply / Breaking speed - Update factor
     }
 
     // CraftBukkit start - add a world/entity limited version
@@ -1018,7 +_,7 @@
         // needs to be done because the ServerPlayer instance is being reused on respawn instead of getting replaced like on vanilla
         java.util.Collection<net.minecraft.world.entity.ai.attributes.AttributeInstance> syncableAttributes = player.getAttributes().getSyncableAttributes();
         player.getBukkitEntity().injectScaledMaxHealth(syncableAttributes, true);
-        player.connection.send(new net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket(player.getId(), syncableAttributes));
+        player.connection.send(new net.minecraft.network.protocol.game.ClientboundUpdateAttributesPacket(player.getId(), syncableAttributes, player)); // Spout - Client view / Packet mapping / Block states / Apply / Breaking speed - Pass self player to attributes packet
         // Paper end - send all attributes
         player.refreshEntityData(player); // CraftBukkit - SPIGOT-7218: sync metadata
         player.connection.send(new ClientboundSetHeldSlotPacket(player.getInventory().getSelectedSlot()));
