--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -277,6 +_,7 @@
         this.paperConfigurations.initializeGlobalConfiguration(this.registryAccess());
         this.paperConfigurations.initializeWorldDefaultsConfiguration(this.registryAccess());
         // Paper end - initialize global and world-defaults configuration
+        spout.server.paper.impl.configuration.SpoutConfigurations.get(this.options).initializeWorldDefaultsConfiguration(this.registryAccess()); // Spout - Configuration - Initialize world-defaults configuration
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
@@ -291,6 +_,13 @@
         com.destroystokyo.paper.Metrics.PaperMetrics.startMetrics(); // Paper - start metrics
         com.destroystokyo.paper.VersionHistoryManager.INSTANCE.getClass(); // Paper - load version history now
 
+        // Spout start - Resource pack - Construction - Fire compose event
+        spout.server.paper.impl.resourcepack.construct.ResourcePackConstructionImpl resourcePackConstruction = spout.server.paper.impl.resourcepack.construct.ResourcePackConstructionImpl.get();
+        if (resourcePackConstruction.isEnabled()) {
+            resourcePackConstruction.fireComposeEvent();
+        }
+        // Spout end - Resource pack - Construction - Fire compose event
+
         // this.worldData.setGameType(properties.gameMode.get()); // CraftBukkit - moved to world loading
         LOGGER.info("Default game type: {}", properties.gameMode.get());
         // Paper start - Unix domain socket support
@@ -331,6 +_,8 @@
             if (true) throw new IllegalStateException("Failed to bind to port", e); // Paper - Propagate failed to bind to port error
             return false;
         }
+
+        spout.gamecontent.datadriven.material.enuminjection.MaterialEnumSynchronizer.get().runFromDedicatedServerInit(); // Spout - Material injection - Implementation part - Run on server start, before Bukkit plugins are loaded
 
         // CraftBukkit start
         this.server.loadPlugins();
