--- a/src/main/java/io/papermc/paper/registry/PaperRegistries.java
+++ b/src/main/java/io/papermc/paper/registry/PaperRegistries.java
@@ -112,8 +_,14 @@
             start(Registries.GAME_EVENT, RegistryKey.GAME_EVENT).craft(GameEvent.class, CraftGameEvent::new).writable(PaperGameEventRegistryEntry.PaperBuilder::new),
             start(Registries.STRUCTURE_TYPE, RegistryKey.STRUCTURE_TYPE).craft(StructureType.class, CraftStructureType::new).build(),
             start(Registries.MOB_EFFECT, RegistryKey.MOB_EFFECT).craft(PotionEffectType.class, CraftPotionEffectType::new).build(),
-            start(Registries.BLOCK, RegistryKey.BLOCK).craft(BlockType.class, CraftBlockType::new).build(),
-            start(Registries.ITEM, RegistryKey.ITEM).craft(ItemType.class, CraftItemType::new).build(),
+            // Spout start - More data-driven - Paper registry API - Types - Implementation part - Create Paper registries
+            start(spout.gamecontent.datadriven.blocktype.BuiltInBlockTypeRegistryKey.BLOCK_TYPE, io.papermc.paper.registry.RegistryKey.BLOCK_TYPE_TYPE).craft(spout.gamecontent.datadriven.blocktype.SpoutBlockType.class, spout.api.gamecontent.datadriven.blocktype.BlockTypeTypeImpl::of).writable(spout.api.gamecontent.datadriven.blocktype.registry.BlockTypeTypeRegistryEntryImpl.BuilderImpl::new).delayed(),
+            start(spout.gamecontent.datadriven.itemtype.BuiltInItemTypeRegistryKey.ITEM_TYPE, io.papermc.paper.registry.RegistryKey.ITEM_TYPE_TYPE).craft(spout.gamecontent.datadriven.itemtype.SpoutItemType.class, spout.api.gamecontent.datadriven.itemtype.ItemTypeTypeImpl::of).writable(spout.api.gamecontent.datadriven.itemtype.registry.ItemTypeTypeRegistryEntryImpl.BuilderImpl::new).delayed(),
+            // Spout end - More data-driven - Paper registry API - Types - Implementation part - Create Paper registries
+            // Spout start - More data-driven - Paper registry API - Implementation part - Make writable and delayed
+            start(Registries.BLOCK, RegistryKey.BLOCK).craft(BlockType.class, CraftBlockType::new).writable(spout.api.gamecontent.datadriven.block.registry.BlockTypeRegistryEntryImpl.BuilderImpl::new).delayed(),
+            start(Registries.ITEM, RegistryKey.ITEM).craft(ItemType.class, CraftItemType::new).writable(spout.api.gamecontent.datadriven.item.registry.ItemTypeRegistryEntryImpl.BuilderImpl::new).delayed(),
+            // Spout end - More data-driven - Paper registry API - Implementation part - Make writable and delayed
             start(Registries.VILLAGER_PROFESSION, RegistryKey.VILLAGER_PROFESSION).craft(Villager.Profession.class, CraftVillager.CraftProfession::new).build(),
             start(Registries.VILLAGER_TYPE, RegistryKey.VILLAGER_TYPE).craft(Villager.Type.class, CraftVillager.CraftType::new).build(),
             start(Registries.MAP_DECORATION_TYPE, RegistryKey.MAP_DECORATION_TYPE).craft(MapCursor.Type.class, CraftMapCursor.CraftType::new).build(),
@@ -156,6 +_,18 @@
             start(Registries.POTION, RegistryKey.POTION).apiOnly(PaperSimpleRegistry::potion),
             start(Registries.MEMORY_MODULE_TYPE, RegistryKey.MEMORY_MODULE_TYPE).apiOnly(() -> org.bukkit.Registry.MEMORY_MODULE_TYPE)
             // End generate - RegistryDefinitions
+
+            // Spout start - Define additional registries
+            ,
+            start(spout.gamecontent.datadriven.material.enuminjection.rewrite.registry.MaterialNameRewriterRegistryKey.MATERIAL_NAME_REWRITER, spout.api.SpoutRegistryKey.MATERIAL_NAME_REWRITER).craft(spout.api.gamecontent.datadriven.common.enuminjection.rewrite.EnumNameRewriter.class, spout.api.gamecontent.datadriven.common.enuminjection.rewrite.EnumNameRewriterImpl::new).writable(spout.api.gamecontent.datadriven.common.enuminjection.rewrite.registry.EnumNameRewriterRegistryEntryImpl.Builder::new).delayed(),
+            start(spout.gamecontent.datadriven.serversidetranslation.registry.ServersideTranslationRegistryKey.SERVERSIDE_TRANSLATION, spout.api.SpoutRegistryKey.SERVERSIDE_TRANSLATION).craft(spout.api.gamecontent.datadriven.serversidetranslation.ServersideTranslation.class, spout.api.gamecontent.datadriven.serversidetranslation.ServersideTranslationImpl::new).writable(spout.api.gamecontent.datadriven.serversidetranslation.registry.ServersideTranslationRegistryEntryImpl.Builder::new).delayed(),
+            start(spout.clientview.model.awarenesslevel.BuiltInAwarenessLevelRegistryKey.AWARENESS_LEVEL, spout.api.SpoutRegistryKey.AWARENESS_LEVEL).craft(spout.api.clientview.model.awarenesslevel.AwarenessLevel.class, spout.api.clientview.model.awarenesslevel.AwarenessLevelImpl::new).build(),
+            start(spout.clientview.packetmapping.component.registry.ComponentMappingRegistryKey.COMPONENT_MAPPING, spout.api.SpoutRegistryKey.COMPONENT_MAPPING).craft(spout.api.clientview.packetmapping.component.ComponentMapping.class, spout.api.clientview.packetmapping.component.ComponentMappingImpl::new).writable(spout.api.clientview.packetmapping.component.registry.ComponentMappingRegistryEntryImpl.Builder::new).delayed(),
+            start(spout.clientview.packetmapping.blockstate.macro.type.registry.BuiltInBlockStateMappingMacroTypeRegistryKey.BLOCK_STATE_MAPPING_MACRO_TYPE, spout.api.SpoutRegistryKey.BLOCK_STATE_MAPPING_MACRO_TYPE).craft(spout.api.clientview.packetmapping.blockstate.macro.type.BlockStateMappingMacroType.class, spout.api.clientview.packetmapping.blockstate.macro.type.BlockStateMappingMacroTypeImpl::new).build(),
+            start(spout.clientview.packetmapping.blockstate.macro.registry.BlockStateMappingMacroRegistryKey.BLOCK_STATE_MAPPING_MACRO, spout.api.SpoutRegistryKey.BLOCK_STATE_MAPPING_MACRO).craft(spout.api.clientview.packetmapping.blockstate.macro.BlockStateMappingMacro.class, spout.api.clientview.packetmapping.blockstate.macro.BlockStateMappingMacroImpl::new).writable(spout.api.clientview.packetmapping.blockstate.macro.registry.BlockStateMappingMacroRegistryEntryImpl.Builder::new).delayed(),
+            start(spout.clientview.packetmapping.blockstate.registry.BlockStateMappingRegistryKey.BLOCK_STATE_MAPPING, spout.api.SpoutRegistryKey.BLOCK_STATE_MAPPING).craft(spout.api.clientview.packetmapping.blockstate.BlockStateMapping.class, spout.api.clientview.packetmapping.blockstate.BlockStateMappingImpl::create).writable(spout.api.clientview.packetmapping.blockstate.registry.BlockStateMappingRegistryEntryImpl.Builder::new).delayed(),
+            start(spout.clientview.packetmapping.itemstack.registry.ItemStackMappingRegistryKey.ITEM_STACK_MAPPING, spout.api.SpoutRegistryKey.ITEM_STACK_MAPPING).craft(spout.api.clientview.packetmapping.itemstack.ItemStackMapping.class, spout.api.clientview.packetmapping.itemstack.ItemStackMappingImpl::new).writable(spout.api.clientview.packetmapping.itemstack.registry.ItemStackMappingRegistryEntryImpl.Builder::new).delayed()
+            // Spout end - Define additional registries
         );
         final Map<RegistryKey<?>, RegistryEntry<?, ?>> byRegistryKey = new IdentityHashMap<>(REGISTRY_ENTRIES.size());
         final Map<ResourceKey<?>, RegistryEntry<?, ?>> byResourceKey = new IdentityHashMap<>(REGISTRY_ENTRIES.size());
