--- a/net/minecraft/resources/RegistryDataLoader.java
+++ b/net/minecraft/resources/RegistryDataLoader.java
@@ -147,6 +_,15 @@
         new RegistryDataLoader.RegistryData<>(Registries.DECORATED_POT_PATTERN, DecoratedPotPattern.CODEC),
         new RegistryDataLoader.RegistryData<>(Registries.BLOCK_TRANSFORMER, BlockTransformer.DIRECT_CODEC),
         new RegistryDataLoader.RegistryData<>(Registries.BLOCK_STATE_PROVIDER, BlockStateProvider.DIRECT_CODEC)
+        // Spout start - Load additional registries
+        ,
+        new RegistryDataLoader.RegistryData<>(spout.gamecontent.datadriven.material.enuminjection.rewrite.registry.MaterialNameRewriterRegistryKey.MATERIAL_NAME_REWRITER, spout.gamecontent.datadriven.common.enuminjection.rewrite.EnumNameRewriter.codec()),
+        new RegistryDataLoader.RegistryData<>(spout.gamecontent.datadriven.serversidetranslation.registry.ServersideTranslationRegistryKey.SERVERSIDE_TRANSLATION, spout.gamecontent.datadriven.serversidetranslation.ServersideTranslation.CODEC),
+        new RegistryDataLoader.RegistryData<>(spout.clientview.packetmapping.component.registry.ComponentMappingRegistryKey.COMPONENT_MAPPING, spout.clientview.packetmapping.component.ComponentMapping.CODEC),
+        new RegistryDataLoader.RegistryData<>(spout.clientview.packetmapping.blockstate.macro.registry.BlockStateMappingMacroRegistryKey.BLOCK_STATE_MAPPING_MACRO, spout.clientview.packetmapping.blockstate.macro.type.BlockStateMappingMacroType.MACRO_CODEC.codec()),
+        new RegistryDataLoader.RegistryData<>(spout.clientview.packetmapping.blockstate.registry.BlockStateMappingRegistryKey.BLOCK_STATE_MAPPING, spout.clientview.packetmapping.blockstate.BlockStateMapping.CODEC),
+        new RegistryDataLoader.RegistryData<>(spout.clientview.packetmapping.itemstack.registry.ItemStackMappingRegistryKey.ITEM_STACK_MAPPING, spout.clientview.packetmapping.itemstack.ItemStackMapping.CODEC)
+        // Spout end - Load additional registries
     );
     public static final List<RegistryDataLoader.RegistryData<?>> DIMENSION_REGISTRIES = List.of(
         new RegistryDataLoader.RegistryData<>(Registries.LEVEL_STEM, LevelStem.CODEC)
