--- a/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
+++ b/net/minecraft/network/protocol/game/ClientboundLevelChunkWithLightPacket.java
@@ -41,9 +_,10 @@
         final LevelLightEngine lightEngine,
         final @Nullable BitSet skyChangedLightSectionFilter,
         final @Nullable BitSet blockChangedLightSectionFilter
+        , final net.minecraft.server.level.ServerPlayer player // Spout start - Client view / Packet mapping / Block states / Apply / Chunk packets - Pass the player
     ) {
         // Paper start - Anti-Xray
-        this(levelChunk, lightEngine, skyChangedLightSectionFilter, blockChangedLightSectionFilter, true);
+        this(levelChunk, lightEngine, skyChangedLightSectionFilter, blockChangedLightSectionFilter, true, player); // Spout start - Client view / Packet mapping / Block states / Apply / Chunk packets - Pass the player
     }
     public ClientboundLevelChunkWithLightPacket(
         final LevelChunk levelChunk,
@@ -51,10 +_,11 @@
         final @Nullable BitSet skyChangedLightSectionFilter,
         final @Nullable BitSet blockChangedLightSectionFilter,
         final boolean modifyBlocks
+        , final net.minecraft.server.level.ServerPlayer player // Spout start - Client view / Packet mapping / Block states / Apply / Chunk packets - Pass the player
     ) {
         // Paper end - Anti-Xray
         ChunkPos chunkPos = levelChunk.getPos();
-        io.papermc.paper.antixray.ChunkPacketInfo<net.minecraft.world.level.block.state.BlockState> chunkPacketInfo = modifyBlocks ? levelChunk.getLevel().chunkPacketBlockController.getChunkPacketInfo(levelChunk) : null; // Paper - Anti-Xray
+        io.papermc.paper.antixray.ChunkPacketInfo<net.minecraft.world.level.block.state.BlockState> chunkPacketInfo = modifyBlocks ? levelChunk.getLevel().chunkPacketBlockController.getChunkPacketInfo(levelChunk) : new io.papermc.paper.antixray.ChunkPacketInfo<>(levelChunk); // Paper - Ant-Xray // Spout - Client view / Packet mapping / Block states / Apply / Chunk packets - Always create ChunkPacketInfo
         this(
             chunkPos.x(),
             chunkPos.z(),
@@ -67,6 +_,7 @@
             this.setReady(false);
         }
         levelChunk.getLevel().chunkPacketBlockController.modifyBlocks(this, chunkPacketInfo);
+        this.chunkData.spoutBlockStateInChunkPacketMapper = new spout.clientview.packetmapping.blockstate.apply.chunk.BlockStateInChunkPacketMapper(this, levelChunk, player, chunkPacketInfo);
         // Paper end - Anti-Xray - Modify blocks
     }
 
